scripts/linters/pre_commit_linter.py

```diff
@@ -422,6 +422,23 @@ def _print_complete_summary_of_errors(all_messages):
             python_utils.PRINT(message)
 
 
+def _get_task_output(task, semaphore):
+    """Returns output of running tasks.
+
+    Args:
+        task: object(TestingTaskSpec). The task object to get output of linter.
+        semaphore: threading.Semaphore. The object that controls how many tasks
+            can run at any time.
+
+    Returns:
+        messages: list(str). List of linter messages.
+    """
+    semaphore.acquire()
```

Comment:
Move the acquire outside.

Reply:
Done.

Reply:
@kevinlee12 could you please tell me the reason why we need to move the acquire outside.

Reply:
This is so that the function doesn’t even get put on the memory heap.

Reply:
Thanks :)